airflow/models/dagbag.py

```diff
@@ -284,7 +284,10 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
             self.log.exception(e)
             return []
 
-        if not zipfile.is_zipfile(filepath):
+        _, file_ext = os.path.splitext(os.path.split(filepath)[-1])
+        is_zipfile = file_ext != '.py' and zipfile.is_zipfile(filepath)
+
+        if not is_zipfile:
```

Comment:
If you don’t need the first part of the filename, it’s easier to simply

```python
if not filepath.endswith(".py") and zipfile.is_zipfile(filepath):
```

Reply:
I agree with that. I push new commit